services/src/main/java/org/keycloak/forms/login/freemarker/AuthenticationStateCookie.java

```diff
@@ -69,7 +70,7 @@ public static void generateAndSetCookie(KeycloakSession session, RootAuthenticat
         cookie.setRemainingTabs(rootAuthSession.getAuthenticationSessions().keySet());
 
         try {
-            String encoded = JsonSerialization.writeValueAsString(cookie);
+            String encoded = Encode.urlEncode(JsonSerialization.writeValueAsString(cookie));
```

Comment:
Should this be considered a breaking change? Not sure if anyone would consume this cookie aside from us.

Reply:
Nah, all cookies are internal to Keycloak and folks should not rely on them.

Reply:
You and I both know there will be users out there parsing this :stuck_out_tongue: But yeah, this should not be considered any sort of public API that cannot be broken.